src/command/expire/expire.c

```diff
@@ -0,0 +1,680 @@
+                            {
+                                InfoBackupData *backupData = lstGet(archiveIdBackupList, backupListIdx);
+
+                                // If the backup is earlier than or the same as the retention backup and the backup has an
+                                // archive start
+                                if ((strCmp(backupData->backupLabel, archiveRetentionBackup.backupLabel) <= 0) &&
+                                    (backupData->backupArchiveStart != NULL))
+                                {
+                                    ArchiveRange archiveRange =
+                                    {
+                                        .start = strDup(backupData->backupArchiveStart),
+                                        .stop = NULL,
+                                    };
+
+                                    // If this is not the retention backup, then set the stop, otherwise set the expire max to
+                                    // the archive start of the archive to retain
+                                    if (strCmp(backupData->backupLabel, archiveRetentionBackup.backupLabel) != 0)
+                                        archiveRange.stop = strDup(backupData->backupArchiveStop);
+                                    else
+                                        archiveExpireMax = strDup(archiveRange.start);
+
+                                    LOG_DETAIL("archive retention on backup %s, archiveId = %s, start = %s%s",
+                                        strPtr(backupData->backupLabel),  strPtr(archiveId), strPtr(archiveRange.start),
+                                        (archiveRange.stop != NULL ?
+                                            strPtr(strNewFmt(", stop = %s", strPtr(archiveRange.stop))) : ""));
+
+                                    // Add the archive range to the list
+                                    lstAdd(archiveRangeList, &archiveRange);
+                                }
+                            }
+
+                            // Get all major archive paths (timeline and first 32 bits of LSN)
+                            StringList *walPathList =
+                                strLstSort(
+                                    storageListP(
+                                        storageRepo(),
+                                        strNewFmt(STORAGE_REPO_ARCHIVE "/%s", strPtr(archiveId)), .errorOnMissing = true,
+                                        .expression = STRDEF(WAL_SEGMENT_DIR_REGEXP)),
+                                    sortOrderAsc);
+
+                            for (unsigned int walIdx = 0; walIdx < strLstSize(walPathList); walIdx++)
+                            {
+                                String *walPath = strLstGet(walPathList, walIdx);
+                                removeArchive = true;
+
+                                LOG_DEBUG("Expire process, found major WAL path: %s", strPtr(walPath));
```

Comment:
We already have log entries to tell us we are in the expire process so need to repeat that here.  Debug level logging automatically prints the module/function as well.

Ditto for other places where this is output.

Reply:
Removed "Expire process, " from all logs. However, I believe we no longer have them in the expect logs, so I would like to suggest I add testing for them in the unit test.